scripts/dts/edtsdatabase.py

```diff
@@ -0,0 +1,244 @@
+#
+# Copyright (c) 2018 Bobby Noelte
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+from collections.abc import Mapping
+import json
+
+##
+# @brief ETDS Database consumer
+#
+# Methods for ETDS database usage.
+#
+class EDTSConsumerMixin(object):
+    __slots__ = []
+
+    ##
+    # @brief Get all supported device types
+    #
+    # @param none
+    # @return list of supported device types
+    def get_device_types(self):
+        return list(self._edts['device-types'])
+
+    ##
+    # @brief Get all activated compatible devices.
+    #
+    # @param compatibles compatible(s)
+    # @return list of devices that are compatible
+    def get_devices_by_compatible(self, compatibles):
+        devices = dict()
+        if not isinstance(compatibles, list):
+            compatibles = [compatibles, ]
+        for compatible in compatibles:
+            for device_id in self._edts['compatibles'].get(compatible, []):
+                devices[device_id] = self._edts['devices'][device_id]
+        return list(devices.values())
+
+    ##
+    # @brief Get device ids of all activated compatible devices.
+    #
+    # @param compatibles compatible(s)
+    # @return list of device ids of activated devices that are compatible
+    def get_device_ids_by_compatible(self, compatibles):
+        device_ids = dict()
+        if not isinstance(compatibles, list):
+            compatibles = [compatibles, ]
+        for compatible in compatibles:
+            for device_id in self._edts['compatibles'].get(compatible, []):
+                device_ids[device_id] = 1
+        return list(device_ids.keys())
```

Comment:
We need to re-use the code between this function and the previous.

E.g. by having a private function that returns a dict. And then have each of these two call that function and extract either the keys or the values.